akvo/rsr/management/commands/audit_program_aggregation.py

```diff
@@ -0,0 +1,52 @@
+from django.core.management.base import BaseCommand
+from akvo.rsr.usecases import period_update_aggregation as aggregation
+from django.db.models import Q
+from akvo.rsr.models import ProjectHierarchy, IndicatorPeriod
+from akvo.rsr.models.result.utils import QUANTITATIVE, PERCENTAGE_MEASURE, calculate_percentage
+from akvo.utils import ensure_decimal
+
+
+def audit_period_aggregation(period):
+    count = 0
+    value, numerator, denominator = aggregation.sum_updates(period)
+    if period.indicator.measure == PERCENTAGE_MEASURE:
+        contrib_numerator, contrib_denominator = aggregation.sum_contributed_percentage_value(period)
+        numerator = ensure_decimal(numerator) + ensure_decimal(contrib_numerator)
+        denominator = ensure_decimal(denominator) + ensure_decimal(contrib_denominator)
+        value = calculate_percentage(numerator, denominator)
+    else:
+        value = ensure_decimal(value) + aggregation.sum_contributed_unit_value(period)
+
+    try:
+        assert ensure_decimal(period.actual_value) == value
```

Comment:
How many places after the comma are used for comparison? [By default it's 28](https://docs.python.org/3/library/decimal.html?highlight=compare%20decimal#decimal.DefaultContext) which might be a lot and yield invalid results.

Also why use `assert` here instead of `if`?

Reply:
fixed.